tests/unit-ext/test-ext-arg.c

```diff
@@ -211,6 +215,44 @@ test_validator2_handler (const jerry_value_t func_obj_val __attribute__((unused)
   return jerry_create_undefined ();
 } /* test_validator2_handler */
 
+/**
+ * Convert all props of an JS object into native variables.
+ */
+static jerry_value_t
+test_validator_prop_handler (const jerry_value_t func_obj_val __attribute__((unused)), /**< function object */
+                             const jerry_value_t this_val __attribute__((unused)), /**< this value */
+                             const jerry_value_t args_p[], /**< arguments list */
+                             const jerry_length_t args_cnt __attribute__((unused))) /**< arguments length */
+{
+  bool native1 = false;
+  double native2 = 0;
+  double native3 = 3;
+
+  const char *name_p[] = {"prop1", "prop2", "prop3"};
+
+  jerryx_arg_t mapping[] =
+  {
+    jerryx_arg_boolean (&native1, JERRYX_ARG_COERCE, JERRYX_ARG_REQUIRED),
+    jerryx_arg_number (&native2, JERRYX_ARG_COERCE, JERRYX_ARG_REQUIRED),
+    jerryx_arg_number (&native3, JERRYX_ARG_COERCE, JERRYX_ARG_OPTIONAL)
+  };
+
+  jerry_value_t is_ok = jerryx_arg_transform_object_properties (args_p[0],
```

Comment:
I see, it works well if you have to extract properties from the same type of objects over and over again. It might be a burden though, to define a custom transform function, if your object argument is used in a single handler only. I wonder if it might be useful to have something like this available to the user:

```c
typedef struct
{
  const jerry_char_t **name_p,   /**< property name list of the JS object */
  const jerry_length_t name_cnt, /**< count of the name list */
  const jerryx_arg_t *c_arg_p,   /**< points to the array of transformation steps */
  const jerry_length_t c_arg_cnt /**< the count of the `c_arg_p` array */
} jerryx_arg_object_props_t;

jerry_value_t
jerryx_arg_transform_object_props (jerryx_arg_js_iterator_t *js_arg_iter_p,
                                   const jerryx_arg_t *c_arg_p)
{
  jerry_value_t js_arg = jerryx_arg_js_iterator_pop (js_arg_iter_p);
  
  const jerryx_arg_object_props_t *object_props = (const jerryx_arg_object_props_t *) c_arg_p->extra_info;

  return jerryx_arg_transform_object_properties (js_arg, object_props->name_p, object_props->name_cnt, object_props->c_arg_p, object_props->c_arg_cnt);
} /* jerryx_arg_transform_object_props */

static inline jerryx_arg_t
jerryx_arg_object_properties (const jerryx_arg_object_props_t *object_props) /**< pointer to object property mapping */
{
  return (jerryx_arg_t)
  {
    .func = jerryx_arg_transform_object_props,
    .dest = NULL,
    .extra_info = object_props
  };
} /* jerryx_arg_object_properties */
```

It may tempt users to use this function instead of writing specialised custom transform functions, of course.